srcs/Parser.cpp

```diff
@@ -19,44 +19,51 @@ unsigned int	Parser::get_listen(void) const { return (this->listen); }
 
 Error*	Parser::get_error(void) const { return (this->error); }
 
-void	Parser::set_listen(unsigned int listen) { this->listen = listen; }
-
-void	Parser::set_error(Error* error) {
-
-	if ( this->error )
-		delete this->error;
-	this->error = error;
-}
+std::string	Parser::get_root(void) const { return (this->root); }
 
 /* ************************************************************************** */
 /* Other Functions                                                            */
 /* ************************************************************************** */
 
-// This function returns true if the command var is valid and was successfully
-// stored in this->listen. In case of error, the function returns false and
-// this->error stops being nullptr to an instance of the Error class.
+// This function returns true if the command var is valid so that it can be 
+// successfully stored in this->listen. In case of error, the function returns
+// false and this->error stops being nullptr to an instance of the Error class.
 bool	Parser::listen_handler(std::vector<std::string> command) {
 
 	int		port;
 	std::stringstream	s;
 
 	if (command.empty() || command[0] != "listen") {
-		this->set_error(new Error());
+		this->error = new Error();
 		return (false);
 	}
 
 	if (command.size() < 2 ) {
-		this->set_error(new Error());
+		this->error = new Error();
 		return (false);
 	}
 
 	s << command[1];
 	s >> port;
 	if (port <= 0) {
-		this->set_error(new Error());
+		this->error = new Error();
 		return (false);
 	}
 
-	this->set_listen(port);
+	this->listen = port;
+	return (true);
+}
+
+// This function returns true if the command is valid so that it can be
+// successfully stored in this->root. In case of error, the function returns
+// false and this->error stops being nullptr to an instance of the Error class.
+bool	Parser::root_handler(std::vector<std::string> command) {
+
+	std::string root;
```

Comment:
Is std::string root ever used here? (line 62)

Reply:
nop, mb. Can be removed